website/src/blogs/how-to-deploy-a-cassandra-cluster-ring-on-kubernetes-openebs.md

```diff
@@ -0,0 +1,453 @@
+     — Address Load Tokens Owns (effective) Host ID Rack
+     UN 10.2.4.6 83.17 KiB 32 75.6% 3c93c7b7–61a7–4cf1-a407-cb47b1de0763 Rack1-K8Demo
+     UN 10.2.1.6 65.65 KiB 32 59.5% 552569fe-c6df-4edb-a553–9efdcf682fb3 Rack1-K8Demo
+     UN 10.2.2.4 83.12 KiB 32 64.9% 92060271-d8cd-48be-a489-c830a8553462 Rack1-K8Demo
+
+UN means node is **up **and in **normal **state. You will also notice that each node has 32 tokens. This is the default value, in production workloads a good default value for this is 256. See more information [here](http://docs.datastax.com/en/archived/cassandra/2.0/cassandra/architecture/architectureDataDistributeVnodesUsing_c.html).
+
+The **Owns** column suggests the data distribution percentage for the content placed into the Cassandra keyspaces.
+
+#### Create a Test Keyspace with Tables
+
+Identify the IP Address of any of the Cassandra replicas, for example, Cassandra-0. This is available from the output of the `nodetool status` command executed above (10.2.4.6, 10.2.1.6, 10.2.2.4).
+
+**Cqlsh** is a Python-based utility that enables you to execute Cassandra Query Language (CQL). **CQL** is a declarative language that allows users to query Cassandra using semantics similar to SQL.
+
+Install the python-minimal and python-pip apt packages and perform a pip install of Csqlsh using the following commands.
+
+    sudo apt-get install -y python-minimal python-pip
+     pip install cqlsh
+
+Login to the CQL shell using the Cqlsh utility using the following command.
+
+    ubuntu@ip-172–23–1–236:~/openebs/k8s/demo/cassandra$ cqlsh 10.2.4.6 9042 — cqlversion=”3.4.2"
+     Connected to K8Demo at 10.2.4.6:9042.
+     [cqlsh 5.0.1 | Cassandra 3.9 | CQL spec 3.4.2 | Native protocol v4]
+     Use HELP for help.
+     cqlsh>
+
+As I mentioned earlier, you have two placement options while creating a keyspace. You can either use SimpleStrategy or NetworkTopologyStrategy.
+
+You can create a keyspace using SimpleStrategy with replication factor 2 by running the following commands.
+
+    cqlsh> create keyspace ssks with replication = { ‘class’ : ‘SimpleStrategy’ , ‘replication_factor’ : 2 };
+
+I will create a keyspace using NetworkTopologyStrategy by running the following commands.
+
+    cqlsh> create keyspace ntsks with replication = { ‘class’ : NetworkTopologyStrategy’, ‘DC1-K8Demo’ : 1 };
+
+    cqlsh> describe keyspaces;
+
+    ntsks system_schema system_auth system system_distributed system_traces
+
+To use NetworkTopologyStrategy with data centers in a production environment, you need to change the default snitch, **SimpleSnitch** to a network-aware **Ec2Snitch.** You need to define one or more data center names in the snitch properties file, and use those data center names to define the keyspace; otherwise, Cassandra will fail to find a node. You can find the instructions to change the default Snitch [here](https://docs.datastax.com/en/cassandra/2.1/cassandra/operations/ops_switch_snitch.html).
```

Comment:
```suggestion
To use NetworkTopologyStrategy with data centers in a production environment, you need to change the default snitch, **SimpleSnitch** to a network-aware **Ec2Snitch**. You need to define one or more data center names in the snitch properties file, and use those data center names to define the keyspace; otherwise, Cassandra will fail to find a node. You can find the instructions to change the default Snitch [here](https://docs.datastax.com/en/cassandra/2.1/cassandra/operations/ops_switch_snitch.html).
```